test/mocha/10-api.js

```diff
@@ -1,3 +1,505 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const {config} = bedrock;
+const https = require('https');
+const helpers = require('./helpers');
+const mockData = require('./mock.data');
+const {httpClient} = require('@digitalbazaar/http-client');
+const brAccount = require('bedrock-account');
+const brAuthnToken = require('bedrock-authn-token');
+const {authenticator} = require('otplib');
+
+let accounts;
+let actors;
+
+function stubPassportStub(email) {
+  passportStub.callsFake((req, res, next) => {
+    req.user = {
+      actor: actors[email],
+      account: accounts[email].account
+    };
+    next();
+  });
+}
+
+const baseURL = `https://${config.server.host}` +
+  `${config['authn-token-http'].routes.basePath}`;
+const authenticateURL = `https://${config.server.host}` +
+  `${config['authn-token-http'].routes.authenticate}`;
+const registrationURL = `https://${config.server.host}` +
+`${config['authn-token-http'].routes.registration}`;
+
+// use agent to avoid self-signed certificate errors
+const agent = new https.Agent({rejectUnauthorized: false});
```

Comment:
this should use `bedrock-http-agent` and corresponding entries in test.config.js which I see are already included in the test config here.

Reply:
I followed this as a reference. https://github.com/digitalbazaar/bedrock-profile-http/blob/ed90c26c6ce17b593bb08f7f40070c74a6e95524/test/mocha/10-api.js#L31

Is there an example on how to do this differently?

Reply:
Here you go: https://github.com/digitalbazaar/bedrock-edv-storage/blob/abc79792fb7eda7725e582a5c9dd1bf931ce4d0a/test/mocha/16-update.js#L76-L77